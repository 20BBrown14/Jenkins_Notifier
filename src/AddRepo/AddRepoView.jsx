import React from 'react';
import PropTypes from 'prop-types';
import Button from 'terra-button';
import Text from 'terra-text';
import IconSuccess from 'terra-icon/lib/icon/IconSuccess';
import Image from 'terra-image';
import RepoInputComponent from './RepoInputComponent';
import './AddRepoView.scss';
import loadingSpinner from '../assets/loading_Spinner.gif';

const propTypes = {
  /** Error message for input */
  errorMessage: PropTypes.string.isRequired,
  /** If input is invalid */
  formInvalid: PropTypes.bool,
  /** Function for when repo url input field changes */
  URLFieldChange: PropTypes.func.isRequired,
  /** Function for when repo name input field changes */
  nameFieldChanged: PropTypes.func.isRequired,
  /** Help message to display under input field */
  helpMessage: PropTypes.string,
  /** Value of the repo URL input field */
  repoURL: PropTypes.string.isRequired,
  /** Value of the repo name input field */
  repoName: PropTypes.string.isRequired,
  /** Handler for when validate is clicked */
  validateClickHandler: PropTypes.func,
  /** Handler for when confirm is clicked */
  confirmClickHandler: PropTypes.func,
  /** Whether or not the URL has been validated */
  validated: PropTypes.bool,
  /** Whether or not the URL has been confirmed */
  confirmed: PropTypes.bool,
  /** Function for when the cancel button is clicked */
  cancelClickHandler: PropTypes.func.isRequired,
  /** Bool to determine if the app is loading or not */
  isLoading: PropTypes.bool.isRequired,
};

const defaultProps = {
  formInvalid: false,
};

const AddRepoView = (props) => {
  const {
    errorMessage,
    formInvalid,
    URLFieldChange,
    nameFieldChanged,
    helpMessage,
    repoURL,
    validateClickHandler,
    confirmClickHandler,
    repoName,
    validated,
    confirmed,
    cancelClickHandler,
    isLoading,
  } = props;

  return (
    <div className="AddRepoview-Window">
      <Text style={{ margin: '12px 0px 0px 25px' }} fontSize={32}>Track a New Repo</Text>
      {(isLoading) && (
        <Image
          src={loadingSpinner}
          height="60"
          width="60"
          style={{ top: '7px', position: 'absolute' }}
        />
      )}
      <RepoInputComponent
        errorMessage={errorMessage}
        formInvalid={formInvalid}
        URLFieldChange={URLFieldChange}
        nameFieldChanged={nameFieldChanged}
        helpMessage={helpMessage}
        repoURL={repoURL}
        repoName={repoName}
        isLoading={isLoading}
      />
      <Button
        style={{ margin: '0px 5px 0px 25px' }}
        text="Validate"
        onClick={validateClickHandler}
        isDisabled={validated || isLoading}
      />
      <Button
        style={{ margin: '0px 5px 0px 5px' }}
        text="Confirm"
        onClick={confirmClickHandler}
        isDisabled={confirmed || isLoading}
      />
      <Button
        style={{ margin: '0px 5px 0px 5px' }}
        text="Cancel"
        onClick={cancelClickHandler}
      />
      {confirmed && validated &&
        <IconSuccess />}
    </div>
  );
};

AddRepoView.propTypes = propTypes;
AddRepoView.defaultProps = defaultProps;

export default AddRepoView;
