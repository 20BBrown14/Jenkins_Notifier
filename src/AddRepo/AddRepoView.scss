.AddRepoview-Window {
  width: 650px;
  height: 450px;
}