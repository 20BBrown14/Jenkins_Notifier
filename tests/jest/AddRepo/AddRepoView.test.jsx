import React from 'react';
import AddRepoView from '../../../src/AddRepo/AddRepoView';

describe('AddRepoView', () => {
  it('should render a default view', () => {
    const testView = (shallow(
      <AddRepoView
        errorMessage="someMessage"
        repoURL=""
        repoName=""
        URLFieldChange={() => {}}
        nameFieldChanged={() => {}}
        cancelClickHandler={() => {}}
        isLoading={false}
      />,
    ));
    expect(testView).toMatchSnapshot();
  });
  describe('Button clicking', () => {
    let testView;
    let validateClickHandler;
    let cancelClickHandler;
    let confirmClickHandler;
    beforeEach(() => {
      validateClickHandler = jest.fn();
      cancelClickHandler = jest.fn();
      confirmClickHandler = jest.fn();
      testView = (shallow(
        <AddRepoView
          errorMessage="someMessage"
          validateClickHandler={validateClickHandler}
          URLFieldChange={() => {}}
          nameFieldChanged={() => {}}
          confirmClickHandler={confirmClickHandler}
          repoURL=""
          repoName=""
          cancelClickHandler={cancelClickHandler}
          isLoading={false}
        />,
      ));
    });
    describe('Validate button', () => {
      it('Should call validate button handler', () => {
        const buttons = testView.find('Button');
        expect(buttons.length).toEqual(3);
        buttons.at(0).simulate('click');
        expect(validateClickHandler).toHaveBeenCalled();
        expect(validateClickHandler).toHaveBeenCalledTimes(1);
      });
    });
    describe('Confirm button', () => {
      it('should call confirm click handler', () => {
        const buttons = testView.find('Button');
        expect(buttons.length).toEqual(3);
        buttons.at(1).simulate('click');
        expect(confirmClickHandler).toHaveBeenCalled();
        expect(confirmClickHandler).toHaveBeenCalledTimes(1);
      });
    });
    describe('Cancel button', () => {
      it('should call cancel click handler', () => {
        const buttons = testView.find('Button');
        expect(buttons.length).toEqual(3);
        buttons.at(2).simulate('click');
        expect(cancelClickHandler).toHaveBeenCalled();
        expect(cancelClickHandler).toHaveBeenCalledTimes(1);
      });
    });
  });
});
