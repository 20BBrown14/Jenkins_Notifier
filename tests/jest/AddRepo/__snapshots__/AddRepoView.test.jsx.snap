// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRepoView should render a default view 1`] = `
<div
  className="AddRepoview-Window"
>
  <Text
    color="inherit"
    fontSize={32}
    isItalic={false}
    isVisuallyHidden={false}
    style={
      Object {
        "margin": "12px 0px 0px 25px",
      }
    }
  >
    Track a New Repo
  </Text>
  <RepoInputComponent
    URLFieldChange={[Function]}
    errorMessage="someMessage"
    formInvalid={false}
    nameFieldChanged={[Function]}
    repoName=""
    repoURL=""
  />
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    style={
      Object {
        "margin": "0px 5px 0px 25px",
      }
    }
    text="Validate"
    type="button"
    variant="neutral"
  />
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    style={
      Object {
        "margin": "0px 5px 0px 5px",
      }
    }
    text="Confirm"
    type="button"
    variant="neutral"
  />
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    onClick={[Function]}
    style={
      Object {
        "margin": "0px 5px 0px 5px",
      }
    }
    text="Cancel"
    type="button"
    variant="neutral"
  />
</div>
`;
